lib/adequate_serializer/collection.rb

```diff
@@ -29,7 +29,17 @@ def as_json(options = {})
     private
 
     def root_name
-      (root || collection.first.class.name.underscore.parameterize.pluralize).to_sym
+      (root || build_root).to_sym
+    end
+
+    def build_root
+      if collection.respond_to?(:klass)
+        klass = collection.klass
+      else
+        klass = collection.first.class
+      end
+
+      klass == NilClass ? 'nil' : klass.name.underscore.parameterize.pluralize
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.